benchmark-service/benchmark/benchmark.py

```diff
@@ -232,17 +235,22 @@ async def update_commits(app):
     for gh_commit in gh_data:
         sha = gh_commit.get('sha')
         log.info(f'for commit {sha}')
-        file_path = f'gs://{HAIL_BENCHMARK_BUCKET_NAME}/benchmark-test/{sha}.json'
+        file_path = f'{BENCHMARK_RESULTS_PATH}/{sha}.json'
         has_results_file = gs_reader.file_exists(file_path)
-        batches = [b async for b in batch_client.list_batches(q=f'sha={sha} running')]
 
-        if not batches and not has_results_file:
+        if has_results_file:
+            log.info(f'commit {sha} has a results file')
+            batches = [b async for b in batch_client.list_batches(q=f'sha={sha} complete')]
+            batch = batches[-1]
+            assert batch, batches
+        elif [b async for b in batch_client.list_batches(q=f'sha={sha} running')]:
```

Comment:
Actually, this is a better way. Can you double check with dev deploy? Not sure I got the batch result structure correct to extract the state.

```
batches = [b async for b in batch_client.list_batches(q=f'sha={sha}')]
complete_batches = [b for b in batches if b['state'] == 'complete']
running_batches = [b for b in batches if b['state'] == 'running']

if has_results_file:
    batch = complete_batches[-1]
elif running_batches:
    batch = running_batches[-1]
else:
    ...
``` 


if has_results_file:

else:
     batches = [b async for b in batch_client.list_batches(q=f'sha={sha} running')]